hack/lib/integration.sh

```diff
@@ -124,10 +124,25 @@ readonly -f os::integration::configresolver::check_log
 __os_integration_boskos_pid=
 
 # os::integration::boskos::start starts an in-memory boskos server.
+# The name of a valid resource is required to verify that the server has
+# successfully started.
 function os::integration::boskos::start() {
-    local config=$1
+    local config=$1 test_lease=$2
+    local url=http://localhost:8080
+    local curl='curl --request POST --silent --show-error --fail'
     boskos --in_memory --config "$1" &> "${LOG_DIR}/boskos.log" &
     __os_integration_boskos_pid=$!
+    os::log::info "Waiting for boskos to be ready..."
+    if ! os::cmd::try_until_success "${curl} '${url}/acquire?type=${test_lease}&state=free&dest=leased&owner=test'" 30; then
+        os::integration::boskos::stop
```

Comment:
if this fails, will we still see the log?

Reply:
Yes.  The `bash` is a bit inscrutable, but I've tested both error scenarios.

That's the source of the error message in the description:

https://github.com/openshift/ci-tools/blob/a4e5d83ca6233425f558ccf6658da259a2bbc198/hack/lib/integration.sh#L139-L141

If the `stop` function succeeds (we failed to acquire but the server is up), then we get:

```bash
cat "${LOG_DIR}/boskos.log"
os::log::fatal 'failed to acquire the test lease'
```